cranial/messaging/file.py

```diff
@@ -67,6 +70,45 @@ def send(self, address=None, message='', endpoint=None, serde=json,
                 "{} || endpoint: {} || message: {}".format(
                     e, endpoint, message))
 
+    def get_last_id(self, endpoint: str = '', sep: str = '', bucket: str = None,
+                    prefix: str = None):
+        """ Takes an S3 endpoint and the seperator used in naming files
+            gets all the files at the prefix in the given endpoint
+            gets the most recently modified file with an id
+            reads the last row from that file and returns that id as the
+            last_id
+        """
+        if sep == '':
+            raise Exception("Need seperator to parse last_id from file names")
+
```

Comment:
The use of the id in the filename is just a convention in the pipe application. I think we just need to get the last file by date and the last line of that file and parse it, so all we have to pass in here is the bucket, path prefix, and name of the field storing the id value, which should of course default to "id".